unidep/_setuptools_integration.py

```diff
@@ -151,7 +151,8 @@ def get_python_dependencies(
         for paths in local_dependencies.values():
             for path in paths:
                 name = _package_name_from_path(path)
-                dependencies.append(f"{name} @ file://{path.as_posix()}")
+                uri = path.as_posix().replace(" ", "%20")
+                dependencies.append(f"{name} @ file://{uri}")
```

Comment:
You should probably do proper path -> URL translation to handle funky paths like `@this"ISa;LEGITpath` using [`urllib.request.pathname2url()`](https://docs.python.org/3/library/urllib.request.html#urllib.request.pathname2url):

```suggestion
                uri = "file:" + urllib.request.pathname2url(path)
                dependencies.append(f"{name} @ file://{uri}")
```

Reply:
Thanks for the suggestion! I previously tried `urllib.quote` but all Windows builds started to fail.

Reply:
Sorry, I just made a correction. I didn't read pip's use of it carefully enough. It expects a string path and you should ensure that the pathlib path is already absolute before passing it to the function (although I'd guess that to be the case already as otherwise this logic would be blowing up today with relative paths).

Reply:
Yeah it's an absolute path already 😃

Is the edit just the str call?

mypy had already caught it. 

Thank you for taking a look here BTW!